Sources/FoundationEssentials/UUID.swift

```diff
@@ -76,6 +76,50 @@ public struct UUID : Hashable, Equatable, CustomStringConvertible, Sendable {
             hasher.combine(bytes: buffer)
         }
     }
+    
+    /// Generates a new random UUID.
+    ///
+    /// - Parameter generator: The random number generator to use when creating the new random value.
+    /// - Returns: A random UUID.
+    @available(FoundationPreview 6.2, *)
+    public static func random(
+        using generator: inout some RandomNumberGenerator
+    ) -> UUID {
+        let first = UInt64.random(in: .min ... .max, using: &generator)
+        let second = UInt64.random(in: .min ... .max, using: &generator)
+
+        var firstBits = first
+        var secondBits = second
+
+        // Set the version to 4 (0100 in binary)
+        firstBits &= 0xFFFFFFFFFFFF0FFF // Clear the last 12 bits
```

Comment:
Is the code wrong or the comment wrong? This appears to clear bits 12 ... 15.

Reply:
The comment was wrong. This clears bits 48-51 of the UInt64 aligned with the RFC. Example below

10010100 10010010 10111010 11000100 11110011 01010011 **0010**1001 11100111
10010100 10010010 10111010 11000100 11110011 01010011 **0000**1001 11100111

Reply:
Let's use binary literals maybe? That way your comment here becomes the code :)